src/code/z_skin.c

```diff
@@ -1,23 +1,272 @@
+    wTemp.y = mtx->wy;
+    wTemp.z = mtx->wz;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin/func_80138050.s")
+    mtx->wx = 0.0f;
+    mtx->wy = 0.0f;
+    mtx->wz = 0.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin/func_80138228.s")
+    for (vertexEntry = skinVertices; vertexEntry < &skinVertices[modifEntry->vtxCount]; vertexEntry++) {
+        vtx = &vtxBuf[vertexEntry->index];
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin/func_80138258.s")
+        vtx->n.ob[0] = pos->x;
+        vtx->n.ob[1] = pos->y;
+        vtx->n.ob[2] = pos->z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin/func_8013828C.s")
+        sp44.x = vertexEntry->normX;
+        sp44.y = vertexEntry->normY;
+        sp44.z = vertexEntry->normZ;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin/func_801382C4.s")
+        SkinMatrix_Vec3fMtxFMultXYZ(mtx, &sp44, &normal);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin/func_80138300.s")
+        vtx->n.n[0] = normal.x;
+        vtx->n.n[1] = normal.y;
+        vtx->n.n[2] = normal.z;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin/func_8013835C.s")
+    mtx->wx = wTemp.x;
+    mtx->wy = wTemp.y;
+    mtx->wz = wTemp.z;
+}
+
+void Skin_ApplyLimbModifications(GraphicsContext* gfxCtx, Skin* skin, s32 limbIndex, s32 arg3) {
+    s32 modifCount;
+    SkinLimb** skeleton;
+    s32 pad;
+    SkinAnimatedLimbData* data;
+    s32 pad2[5];
+    SkinLimbModif* modif;
+    SkinLimbVtx* vtxEntry;
+    s32 pad3;
+    s32 transformCount;
+    f32 scale;
+    SkinVertex* skinVertices;
+    Vec3f sp88;
+    Vtx* vtxBuf;
+    SkinTransformation* limbTransformations;
+    Vec3f spDC;
+    Vec3f spD0;
+    SkinTransformation* transformationEntry;
+    SkinLimb* limb;
+    SkinLimbModif* modifications;
+
+    OPEN_DISPS(gfxCtx);
+
+    skeleton = (SkinLimb**)Lib_SegmentedToVirtual(skin->skeletonHeader->segment);
+    limb = (SkinLimb*)Lib_SegmentedToVirtual(skeleton[limbIndex]);
+    data = Lib_SegmentedToVirtual(limb->segment);
+    modifications = (SkinLimbModif*)Lib_SegmentedToVirtual(data->limbModifications);
+
+    vtxEntry = &skin->vtxTable[limbIndex];
+
+    vtxBuf = vtxEntry->buf[vtxEntry->index];
+    modifCount = data->limbModifCount;
+
+    for (modif = modifications; modif < &modifications[modifCount]; modif++) {
+        Vec3f spAC;
+        Vec3f spA0;
+
+        skinVertices = (SkinVertex*)Lib_SegmentedToVirtual(modif->skinVertices);
+        limbTransformations = (SkinTransformation*)Lib_SegmentedToVirtual(modif->limbTransformations);
+        transformCount = modif->transformCount;
+
+        if (transformCount == 1) {
+            spAC.x = limbTransformations[0].x;
+            spAC.y = limbTransformations[0].y;
+            spAC.z = limbTransformations[0].z;
+
+            SkinMatrix_Vec3fMtxFMultXYZ(&gSkinLimbMatrices[limbTransformations[0].limbIndex], &spAC, &spDC);
+        } else if (arg3) {
+            transformationEntry = &limbTransformations[modif->transformIndex];
```

Comment:
So it seems like
- if there's one, it will do that
- if not, and arg3, it will do the one corresponding to the index
- otherwise, it'll do all of them, in order?

Reply:
thats the idea, but my main issue with naming this still is the why-- I dont understand what `modif->transformIndex` even is and why you would want to single one specific one out. Which is why this variable is still unnamed over in oot